app/services/fetch_link_card_service.rb

```diff
@@ -1,9 +1,15 @@
 # frozen_string_literal: true
 
 class FetchLinkCardService < BaseService
-  include ActionView::Helpers::TagHelper
-
-  URL_PATTERN = %r{https?://\S+}
+  URL_PATTERN = %r{
```

Comment:
URL_PATTERN is also used in StatusLengthValidator, to treat all URLs as 23 characters length. Same regex is also used in WebUI side to calculate and display character count.

https://github.com/tootsuite/mastodon/blob/bdcc9e2ceb3a359e43775de9b2afd5b1e64eef4a/app/javascript/mastodon/features/compose/util/counter.js#L5

So if you want to change it, you also should change WebUI side too. WebUI allowed some status but server returns an error, will confuses user.